bootloader/bootloadertest/bootloadertest.go

```diff
@@ -70,9 +72,18 @@ func Mock(name, bootdir string) *MockBootloader {
 
 		runKernelImageMockedErrs:     make(map[string]error),
 		runKernelImageMockedNumCalls: make(map[string]int),
+
+		// this is for UC16/UC18 boot var for reboot
+		// for UC20, set to "kernel_status"
+		RebootStatusVar: "snap_mode",
 	}
 }
 
+func (b *MockBootloader) UC20RebootMode() *MockBootloader {
```

Comment:
I'm still not satisfied with this, not the name, nor the mechanics. Not your fault, I think the deeper issue is that we have hard-coded and given responsibilities to the MockBootloader, that actually pertain to boottest. As long as we had only one set of boot vars it was ok but now it's becoming confusing/not very clean.

I have some ideas how to fix it, which I think will make things slightly more verbose but also cleaner hopefully, I don't think we should block this PR on this though.

Please let's rename this to  UC20RunModeRebootReady for now, and put an `// FIXME: this clearly shows mixing of responsibilities here with boottest.`

I'll try to work on cleaning this up as soon as possible.

Reply:
To be clear this is not just a cleanliness concern, as we grow more of this kind of tests, there is a risk of testing the wrong thing, tests passing for the wrong reasons, or testing not realistic states.

Reply:
Renamed